src/engine.py

```
@@ -9304,25 +9365,16 @@ async def _execute_provider_tool(
-            except Exception:
-                pass
-
+            # Determine allowlisted tools for this call (contexts are the source of truth).
+            allowed_tools: list[str] = []
             try:
                 if getattr(session, "context_name", None):
                     ctx_cfg = self.transport_orchestrator.get_context_config(session.context_name)
-                    if ctx_cfg and hasattr(ctx_cfg, "tools") and ctx_cfg.tools is not None:
-                        allowed_tools = list(ctx_cfg.tools or [])
+                    allowed_tools = list(getattr(ctx_cfg, "tools", None) or []) if ctx_cfg else []
             except Exception:
```

Ruff (0.14.10)

Do not catch blind exception: `Exception`

(BLE001)